mayastor/src/bdev/nexus/nexus_bdev_rebuild.rs

```diff
@@ -37,17 +37,24 @@ impl Nexus {
     ) -> Result<Receiver<RebuildState>, Error> {
         trace!("{}: start rebuild request for {}", self.name, name);
 
-        let src_child_name =
-            match self.children.iter().find(|c| c.state == ChildState::Open) {
-                Some(child) => Ok(child.name.clone()),
-                None => Err(Error::NoRebuildSource {
-                    name: self.name.clone(),
-                }),
-            }?;
+        let src_child_name = match self
+            .children
+            .iter()
+            .find(|c| c.state == ChildState::Open && c.name != name)
+        {
+            Some(child) => Ok(child.name.clone()),
+            None => Err(Error::NoRebuildSource {
+                name: self.name.clone(),
+            }),
+        }?;
 
         let dst_child = match self.children.iter_mut().find(|c| c.name == name)
         {
-            Some(c) => Ok(c),
+            Some(c) if c.state == ChildState::Faulted => Ok(c),
+            Some(_) => Err(Error::ChildNotFaulted {
+                child: name.to_owned(),
+                name: self.name.clone(),
+            }),
```

Comment:
Would it be useful to return the child state if not faulted?

Reply:
yeah good idea, Done